pennylane/gradients/gradient_transform.py

```diff
@@ -303,7 +324,21 @@ def jacobian_wrapper(
                 return qjac
 
             kwargs.pop("shots", False)
-            cjac = cjac_fn(*args, **kwargs)
+
+            # Special case where we apply a Jax transform (jacobian e.g.) on the gradient transform and argnums are
+            # defined on the outer transform and therefore on the args.
+            if argnums is None and argnums_ is None and interface == "jax":
+                cjac = qml.transforms.classical_jacobian(
+                    qnode, argnum=qml.math.get_trainable_indices(args), expand_fn=self.expand_fn
+                )(*args, **kwargs)
+            else:
+                if interface == "jax":
+                    argnum_cjac = argnums if argnums is not None else [0]
+                else:
+                    argnum_cjac = None

```

Comment:
doesn't this mean that `argnum_cjac` is always `None` for non-jax computations?

Reply:
Yes it is correct, it is the current way of working for other interfaces, you don't pass the argnum to cjac.

Reply:
it kind of looks like we do [here](https://github.com/PennyLaneAI/pennylane/blob/b36054fbe7bf62f1186f14139d08c8d189a5f8cc/pennylane/gradients/gradient_transform.py#L286), no?

Reply:
It is a recent change that I made for the Jax interface (as a first step to make it work), argnums is always None for other interfaces.

Reply:
https://github.com/PennyLaneAI/pennylane/pull/3776/files#diff-7498bddbc39a804610097ec71b60020e226e1f244f3d4d124faa6be9cfdd1c91R285

Reply:
Simplification done, thanks for pointing this out :+1: